pkg/service/init.go

```diff
@@ -10,86 +10,129 @@ import (
 	"context"
 	"sync"
 
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/handlers"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/startup"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/di"
 	"github.com/gorilla/mux"
 
-	"github.com/edgexfoundry/device-sdk-go/v3/internal/common"
-
 	"github.com/edgexfoundry/device-sdk-go/v3/internal/cache"
+	"github.com/edgexfoundry/device-sdk-go/v3/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/v3/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/v3/internal/controller/http"
+	"github.com/edgexfoundry/device-sdk-go/v3/internal/controller/messaging"
 	"github.com/edgexfoundry/device-sdk-go/v3/internal/provision"
 	"github.com/edgexfoundry/device-sdk-go/v3/pkg/models"
 )
 
 // Bootstrap contains references to dependencies required by the BootstrapHandler.
 type Bootstrap struct {
-	router *mux.Router
+	deviceService *deviceService
+	router        *mux.Router
 }
 
 // NewBootstrap is a factory method that returns an initialized Bootstrap receiver struct.
-func NewBootstrap(router *mux.Router) *Bootstrap {
+func NewBootstrap(ds *deviceService, router *mux.Router) *Bootstrap {
 	return &Bootstrap{
-		router: router,
+		deviceService: ds,
+		router:        router,
 	}
 }
 
-func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer startup.Timer, dic *di.Container) (success bool) {
-	ds.UpdateFromContainer(b.router, dic)
-	ds.ctx = ctx
-	ds.wg = wg
-	ds.controller.InitRestRoutes()
+func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, _ startup.Timer, dic *di.Container) (success bool) {
+	s := b.deviceService
+	s.wg = wg
+	s.ctx = ctx
+	s.lc = bootstrapContainer.LoggingClientFrom(dic.Get)
+	s.autoEventManager = container.ManagerFrom(dic.Get)
```

Comment:
I actually planned to move AutoEventManager to internal directory if possible because user don't need to implement it or use it. May we do this in that change ?

Reply:
you're right, updated.